### ISSUE TITLE
Switch clipboard crate to copypasta

### ISSUE THREAD

**Issue by dbr** [imgui-rs/imgui-rs#623]
Currently the imgui-example code uses the `clipboard` crate, which works fine, but hasn't been touched in about 3 years currently and lacks support for Wayland

There is a fork actively developed for by the alacritty project which is more or less a straight switch out
https://github.com/alacritty/copypasta

Only slight ~issue is the Wayland integration requires access to the window object (e.g `&glutin::window::Window`) so it ties the integration code to a specific, whereas X11 is more lax so doesn't require this

**Comment by reduf** [comment]
There is also an [issue](https://github.com/aweinstock314/rust-clipboard/issues/88) and a [MR](https://github.com/aweinstock314/rust-clipboard/pull/89) which fixes many security issues that is not getting merged due to the crate being abandoned. 
![image](https://user-images.githubusercontent.com/1349199/223660565-daa759fe-4cf9-49df-b479-e80ae2f029f7.png)

It would indeed be good to replace the example, because I assume (I did at least) use it as a starting point.

I'll submit a MR asap.

**Comment by reduf** [comment]
Thanks for the info.

I checked and `x11-clipboard` can't be totally unmaintained, they merged a PR yesterday. I give it to you though, it was the first since July 2022.

Anyway, it could make sense to replace it by `arboard`. One thing were to be careful is to disable the default features as it pulls in even more crates to support images.

**Comment by parasyte** [comment]
You are right, `x11-clipboard` is not totally unmaintained. It is noted in the provided links that it is [primarily in maintenance mode](https://github.com/quininer/x11-clipboard/issues/19) and depends on `xcb` with known unsoundness.

**Comment by parasyte** [comment]
The more I dig in, the crazier it gets! As of `copypasta` 0.8.2, all of the mentioned issues with `x11-clipboard` have been fixed. It no longer depends on the unsound `xcb` crate.

I suspect using `copypasta` is the right thing to do until [`winit` adds clipboard support](https://github.com/rust-windowing/winit/pull/2156) itself.

**Comment by dbr** [comment]
Now merged - info above is interesting, thanks, wasn't aware of arboard nor the plans to integrate clipboard handling directly into winit :partying_face:
